examples/simulcast-sender/main.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * libdatachannel simulcast sender example
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#include "rtc/rtc.hpp"
+
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <utility>
+
+#ifdef _WIN32
+#define _WINSOCK_DEPRECATED_NO_WARNINGS
+#include <WS2tcpip.h>
+#include <io.h>
+#include <winsock2.h>
+#else
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+typedef int SOCKET;
+#endif
+
+
+
+const int BUFFER_SIZE = 2048;
+const int EXTENSION_HEADER_SIZE = 8;
+
+void do_whip(std::shared_ptr<rtc::PeerConnection> pc) {
+	int status;
+	SOCKET client_fd;
+	struct sockaddr_in serv_addr;
+	char buffer[BUFFER_SIZE] = {0};
+	struct addrinfo hints, *servinfo, *p;
+
+	if ((client_fd = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
+		throw std::runtime_error("Failed to create socket for WHIP client");
+	}
+
+	serv_addr.sin_family = AF_INET;
+	serv_addr.sin_port = htons(80);
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = AF_UNSPEC;
+	hints.ai_socktype = SOCK_STREAM;
+	if (getaddrinfo("whip.siobud.com", "80", &hints, &servinfo) != 0) {
+		throw std::runtime_error("Failed to resolve WHIP server");
+	}
+
+	for (p = servinfo; p != NULL; p = p->ai_next) {
+		if ((client_fd = socket(p->ai_family, p->ai_socktype, p->ai_protocol)) == -1) {
+			continue;
+		}
+
+		if (connect(client_fd, p->ai_addr, p->ai_addrlen) == -1) {
+			close(client_fd);
+			continue;
+		}
+
+		break;
+	}
+
+	if (p == NULL) {
+		throw std::runtime_error("Failed to connect to WHIP server");
+	}
+
+	auto description = std::string(pc->localDescription().value());
+
+	std::string request =
+	    std::string("POST /api/whip HTTP/1.1\r\n") + "Host: whip.siobud.com\r\n" +
+	    "Authorization: Bearer seanTest\r\n" + "Content-Type: application/sdp\r\n" +
+	    "Content-Length: " + std::to_string(description.length()) + "\r\n\r\n" + description;
+	if (send(client_fd, request.c_str(), strlen(request.c_str()), 0) == -1) {
+		throw std::runtime_error("Failed to send offer to WHIP server");
+	}
+
+	if (read(client_fd, buffer, BUFFER_SIZE) == -1) {
```

Comment:
I think there is no guarantee this works as `read()` returns as soon as 1 byte or more could be read. By chance the HTTP answer is typically received in a single TCP segment, but you should not rely on the behavior. It would be better practice to concatenate received data into a string until `read()` returns 0.

Reply:
@paullouisageneau I was thinking we probably shouldn't commit this example at all. I just wanted it as a demonstration for you.

https://b.siobud.com probably won't be up forever. Don't want to break your examples if I change anything!

Reply:
OK, makes sense!